pkg/reconciler/managed/reconciler.go

```diff
@@ -1159,45 +1154,48 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (resu
-		// for example if our provider credentials don't have access to update
-		// it. If this is the first time we encounter this issue we'll be
-		// requeued implicitly when we update our status with the new error
-		// condition. If not, we requeue explicitly, which will trigger backoff.
-		log.Debug("Cannot update external resource")
-		record.Event(managed, event.Warning(reasonCannotUpdate, err))
-		managed.SetConditions(xpv1.ReconcileError(errors.Wrap(err, errReconcileUpdate)))
-		return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
-	}
+		if _, err := r.managed.PublishConnection(ctx, managed, update.ConnectionDetails); err != nil {
+			// If this is the first time we encounter this issue we'll be requeued
+			// implicitly when we update our status with the new error condition. If
+			// not, we requeue explicitly, which will trigger backoff.
+			log.Debug("Cannot publish connection details", "error", err)
+			record.Event(managed, event.Warning(reasonCannotPublish, err))
+			managed.SetConditions(xpv1.Updating(), xpv1.ReconcileError(err))
+			return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
+		}
 
-	if _, err := r.managed.PublishConnection(ctx, managed, update.ConnectionDetails); err != nil {
-		// If this is the first time we encounter this issue we'll be requeued
-		// implicitly when we update our status with the new error condition. If
-		// not, we requeue explicitly, which will trigger backoff.
-		log.Debug("Cannot publish connection details", "error", err)
-		record.Event(managed, event.Warning(reasonCannotPublish, err))
-		managed.SetConditions(xpv1.ReconcileError(err))
+		// We've successfully updated our external resource. In many cases the
+		// updating process takes a little time to finish. We requeue explicitly
+		// order to observe the external resource to determine whether it's
+		// ready for use.
+		log.Debug("Successfully requested update of external resource")
+		record.Event(managed, event.Normal(reasonUpdated, "Successfully requested update of external resource"))
+		managed.SetConditions(xpv1.Updating(), xpv1.ReconcileSuccess())
 		return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
 	}
 
-	// We've successfully updated our external resource. Per the below issue
-	// nothing will notify us if and when the external resource we manage
-	// changes, so we requeue a speculative reconcile after the specified poll
-	// interval in order to observe it and react accordingly.
+	// We did not need to create, update, or delete our external resource.
+	// Per the below issue nothing will notify us if and when the external
+	// resource we manage changes, so we requeue a speculative reconcile
+	// after the specified poll interval in order to observe it and react
+	// accordingly.
```

Comment:
This is the comment that was used for the `if observation.ResourceUpToDate` case.

We moved it to the bottom as it seems to work well for the last resort case, which is when there is nothing we need to do.